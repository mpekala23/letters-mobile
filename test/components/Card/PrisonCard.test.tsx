import React from "react";
import { PrisonCard } from "@components";
import { fireEvent, render, toJSON } from "@testing-library/react-native";

const setup = (propOverrides = {}) => {
  const props = Object.assign(
    {
      name: "Name",
      type: "StatePrison",
      address: "Address",
      onPress: jest.fn(),
    },
    propOverrides
  );
  return {
    ...render(<PrisonCard {...props} />),
    props,
  };
};

describe("Prison Card component", () => {
  it("should match snapshot", () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it("should fire onPress() on a press", () => {
    const { props, getByText } = setup();
    fireEvent.press(getByText("Name"));
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });

  it("should display name", () => {
    const { getByText } = setup();
    expect(getByText("Name")).toBeDefined();
  });

  it("should display prison type", () => {
    const { getByText } = setup();
    expect(getByText("State Prison")).toBeDefined();
  });

  it("should display address", () => {
    const { getByText } = setup();
    expect(getByText("Address")).toBeDefined();
  });
});
