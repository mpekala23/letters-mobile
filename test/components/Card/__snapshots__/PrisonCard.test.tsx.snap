// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prison Card component should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 6,
          "marginVertical": 6,
          "padding": 15,
          "width": "100%",
        },
        Object {
          "elevation": 4,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 2.62,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#282828",
          "fontSize": 28,
        }
      }
    >
      Name
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#6E757C",
            "fontSize": 18,
          },
          Object {
            "marginVertical": 6,
          },
        ]
      }
    >
      State Prison
    </Text>
    <Text
      style={
        Object {
          "color": "#6E757C",
          "fontSize": 18,
        }
      }
    >
      Address
    </Text>
  </TouchableOpacity>
</View>
`;
